objects/src/accounts/vault.rs

```diff
@@ -233,10 +232,7 @@ impl ToAdviceInputs for AccountVault {
 
         // populate advice map with tiered merkle tree leaf nodes
         self.asset_tree.upper_leaves().for_each(|(node, key, value)| {
-            // TODO: temporary hack - assume the node is interted at depth 16 and compute remaining key accordingly
-            let mut key = Word::from(key);
-            key[3] = ((key[3].as_int() << 16) >> 16).into();
-            target.insert_into_map(*node, key.into_iter().chain(value).collect());
+            target.insert_into_map(*node, (*key).into_iter().chain(value).collect());
         })
```

Comment:
We should probably add a TODO (or maybe create an issue) to add handling of bottom leaves in the future as well.